index.js

```diff
@@ -2,58 +2,98 @@
      * @method constructor
      * @param  {Object}         config                      Object with executor and ecosystem
      * @param  {String}         [config.defaultPlugin]      Optional default executor
      * @param  {Object}         [config.ecosystem]          Optional object with ecosystem values
      * @param  {Array|Object}   config.executor             Array of executors to load or a single executor object
+     * @param  {Object}         config.redisConnection      Connection details for redis
      * @param  {String}         config.executor[x].name     Name of the executor NPM module to load
      * @param  {String}         config.executor[x].options  Configuration to construct the module with
      */
     constructor(config = {}) {
-        const ecosystem = config.ecosystem || {};
         const executorConfig = config.executor;
-        const defaultPlugin = config.defaultPlugin;
+        const ecosystem = config.ecosystem || {};
 
         if (!executorConfig || !(Array.isArray(executorConfig)) || executorConfig.length === 0) {
             throw new Error('No executor config passed in.');
         }
 
         super();
 
-        executorConfig.forEach((plugin) => {
-            let ExecutorPlugin;
+        this.ecosystem = config.ecosystem || {};
+        this.defaultPlugin = config.defaultPlugin;
 
-            try {
-                // eslint-disable-next-line global-require, import/no-dynamic-require
-                ExecutorPlugin = require(`screwdriver-executor-${plugin.name}`);
-            } catch (err) {
-                console.error(err);
+        this.executorConfigs = {};
+        executorConfig.forEach((plugin) => {
+            this.executorConfigs[plugin.name] = Object.assign({ ecosystem }, plugin.options);
 
-                return;
+            if (!this.defaultPlugin) {
+                this.defaultPlugin = plugin.name;
             }
+        });
+
+        // eslint-disable-next-line new-cap
+        this.queue = new Resque.queue({ connection: config.redisConnection });
 
-            const options = Object.assign({ ecosystem }, plugin.options); // Add ecosystem to executor options
-            const executorPlugin = new ExecutorPlugin(options);
+        this.connected = false;
+        this.queue.connect(() => {
+            this.connected = true;
+        });
+    }
+
+    /**
+     * Get the full config for a job
+     * @method getFullConfig
+     * @param {Object} buildConfig  build configuration
+     * @return {Object}
+     */
+    getFullConfig(buildConfig) {
+        const annotations = buildConfig.annotations || {};
+        const executorType = getPluginName(annotations[ANNOTATION_EXECUTOR_TYPE])
+            || this.defaultPlugin;
+        const options = {
+            buildConfig,
+            executor: {}
+        };
+
+        options.executor.name = executorType;
+        options.executor.options = this.executorConfigs[executorType];
+
+        return options;
```

Comment:
@FenrirUnbound What @minz1027  means is validate the variable `options` to have `executor.name`, `executor.options`, and `buildConfig`. 

It's a little confusing since there is variable `options` and the key `executor.options`. Why not rename the variable to `config`? @ian-fox